ls1intum/Artemis
# Improve Plantuml performance for large courses or exams by using a caching mechanism

### Checklist

### Motivation and Context
Cache plantUml diagrams for programming exercise problem statements to improve the server performance in large courses or exams  

### Steps for Testing
1. Open a programming exercises in the course overview --> details view
2. Open it in the online code editor (for students)
3. Open it in the edit in editor (instructors) with different participations (template, solution, own participation)
4. Edit the programming exercise normally, change the UML diagram and switch between edit and preview
--> in all cases make sure the UML diagrams in the problem statement (potentially in the preview mode) is rendered correctly.
